modules/govuk_env_sync/files/govuk_env_sync.sh

```diff
@@ -270,6 +274,57 @@ function get_timestamp_rsync {
   | grep -o '[0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\}T[0-9]\{2\}:[0-9]\{2\}:[0-9]\{2\}')"
 }
 
+function postprocess_signon_production {
+  source_domain_query="SELECT home_uri FROM oauth_applications WHERE name='Publisher';"
+  source_domain=$(echo "${source_domain_query}" |\
+                  sudo -i mysql --database=signon_production | grep publisher | sed s#https://publisher.##g)
+  # local_domain comes from env.d/LOCAL_DOMAIN (see above).
+
+  update_home_uri_query=
+    "UPDATE oauth_applications\
+     SET home_uri = REPLACE(home_uri, '${source_domain}', '${local_domain}')\
+     WHERE home_uri LIKE '%${source_domain}%'"
+
+  echo "${update_home_uri_query}" | sudo -i mysql --database=signon_production
+
+  update_redirect_uri_query=
+    "UPDATE oauth_applications\
+     SET redirect_uri = REPLACE(redirect_uri, '${source_domain}', '${local_domain}')\
+     WHERE redirect_uri LIKE '%${source_domain}%'"
+
+  echo "${update_redirect_uri_query}" | sudo -i mysql --database=signon_production
+}
+
+function postprocess_router {
+  source_domain=$(mongo --quiet --eval \
+    "db = db.getSiblingDB(\"${database}\"); \
+    db.backends.distinct( \"backend_url\", { \"backend_id\": \"static\" });" \
+    | sed s#https://static.##g | tr -d '/')
+
+  # local_domain comes from env.d/LOCAL_DOMAIN (see above).
+
+  licensify_domain="${local_domain//govuk.digital/publishing.service.gov.uk}"
+
+  mongo --quiet --eval \
+    "db = db.getSiblingDB(\"${database}\"); \
+    db.backends.find( { \"backend_id\": { \$ne: \"licensify\" } } ).forEach( \
+      function(b) { b.backend_url = b.backend_url.replace(\".${source_domain}\", \".${local_domain}\"); \
+    db.backends.save(b); } ); \
+    db.backends.find( { \"backend_id\": \"licensify\" } ).forEach( \
+      function(b) { b.backend_url = b.backend_url.replace(\".${source_domain}\", \".${licensify_domain}\"); \
+    db.backends.save(b); } );"
+}
```

Comment:
One possible alternative to this would be to run the `register_backends` Rake task [1] in the Content Store. That would update the backends in the Router API, from the Content Store, meaning that you wouldn't need to touch the database directly.

1: https://github.com/alphagov/content-store/blob/master/lib/tasks/register_backends.rake

Reply:
Hi @cbaines, if you can live with it, I rather update the database directly, as otherwise I have to enable the sync to talk to the content_store somehow.
Having had to add in this post processing stuff is really not ideal, but I would like to move on from the data sync soon (and keep things in one place - We have penciled in a day for a final refactor, though).

Reply:
I'm not too fussed. Keep in mind though that other parts of the data sync currently rely on running rake tasks, e.g. https://github.com/alphagov/govuk-puppet/blob/master/modules/govuk_jenkins/templates/jobs/data_sync_complete_integration.yaml.erb . In my mind, this Content Store action could be done similarly.

Reply:
I will keep this in mind - We were talking to @benthorner earlier about potentially triggering downstream tasks via Jenkins (to avoid giving govuk-backup any direct access and execution privileges..) will continue to think about what to do..